applications/serial_lte_modem/doc/TCPIP_AT_commands.rst

```diff
@@ -0,0 +1,1541 @@
+
+Set command
+-----------
+
+The set command allows you to open or close a socket.
+
+Syntax
+~~~~~~
+
+::
+
+   #XSOCKET=<op>[,<type>,<role>[,<sec_tag>]]
+
+* The ``<op>`` parameter can accept one the following values:
+
+  * ``0`` - Open
+  * ``1`` - Close
+
+* The ``<sec_tag>`` parameter is used in this AT command to indicate to the modem which credentials to use.
+  It is associated with the certificate or PSK.
+  Specifying the ``<sec_tag>`` is mandatory when opening a socket.
+  When TLS/DTLS is expected, the credentials should be stored on the modem side by ``AT%XCMNG`` or by the Nordic nRF Connect/LTE Link Monitor tool.
+  The modem needs to be in the offline state.
+  The DTLS server is not supported.
+
+* The ``<type>`` parameter value depends on the presence of the <sec_tag> parameter.
+  When the ``<sec_tag>`` is not specified:
+
+  * 1: SOCK_STREAM for TCP
+  * 2: SOCK_DGRAM for UDP
+
+  When the ``<sec_tag>`` is specified:
+
+  * 1: SOCK_STREAM for TLS
+  * 2: SOCK_DGRAM for DTLS
+
+* The ``<role>`` parameter can accept one the following values:
+
+  * ``0`` - Client
+  * ``1`` - Server
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XSOCKET: <handle>[,<type>,<protocol>]
+
+* The ``<handle>`` value is an integer and it can be interpreted as follows:
+
+  * Positive - The socket opened successfully.
+  * Negative - The socket failed to open.
+  * ``0`` - The socket closed successfully.
+
+* The ``<type>`` parameter value depends on the presence of the <sec_tag> parameter.
+  When the ``<sec_tag>`` is not specified:
+
+  * 1: SOCK_STREAM for TCP
+  * 2: SOCK_DGRAM for UDP
+
+  When the ``<sec_tag>`` is specified:
+
+  * 1: SOCK_STREAM for TLS
+  * 2: SOCK_DGRAM for DTLS
+
+* The ``<protocol>`` value is present only in the response to a request to open the socket.
+  It can be one of the following:
+
+  * IPPROTO_TCP(6)
+  * IPPROTO_UDP(17)
+  * IPPROTO_TLS_1_2(258)
+  * IPPROTO_DTLS_1_2(273)
+
+Unsolicited notification
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+   #XSOCKET: <error> closed
+
+The ``<error>`` value is a negative integer, and it represents the error value according to the standard POSIX errorno.
+
+Examples
+~~~~~~~~
+
+::
+
+   AT#XSOCKET=1,1,0
+   #XSOCKET: 3, 6, 0
+   OK
+   AT#XSOCKET=1,2,0
+   #XSOCKET: 3, 17, 0
+   OK
+   AT#XSOCKET=0
+   #XSOCKET: 0, closed
+   OK
+   at#xsocket=1,1,0,16842755
+   #XSOCKET: 2, 1, 0, 258
+   OK
+   at#xsocket=1,2,0,16842755
+   #XSOCKET: 2, 2, 0, 273
+   OK
+
+Read command
+------------
+
+The read command allows you to check the socket handle.
+
+Syntax
+~~~~~~
+
+::
+
+   #XSOCKET?
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XSOCKET: <handle>[,<protocol>, <role>]
+
+* The ``<handle>`` value is an integer and it can be interpreted as follows:
+
+  * Positive - The socket is valid.
+  * ``0`` - The socket is closed.
+
+* The ``<protocol>`` value is present only in the response to a request to open the socket.
+  It can be one of the following:
+
+  * IPPROTO_TCP(6)
```

Comment:
Is the protocol value actually the one mentioned in the parenthesis?

Reply:
yes (https://github.com/zephyrproject-rtos/zephyr/blob/master/include/net/net_ip.h#L62)

Reply:
@junqingzou Should we list all the other protocols not mentioned here but listed in the header file?

Reply:
We only support 4 of them, TCP, UDP, TLS, DTLS. No need to list others I would say